lib/core/config.js

```diff
@@ -231,30 +231,26 @@ Config.prototype.loadContractsConfigFile = function() {
   });
 
   let configFilePath = this._getFileOrOject(this.configDir, 'contracts', 'contracts');
-
+  const autoGas = "auto";
   let newContractsConfig = this._mergeConfig(configFilePath, configObject, this.env);
-  if (newContractsConfig.gas != "auto") {
-      newContractsConfig.gas = utils.getWeiBalanceFromString(newContractsConfig.gas, web3);
+  if (newContractsConfig.gas != autoGas) {
+    newContractsConfig.gas = utils.getWeiBalanceFromString(newContractsConfig.gas, web3);
   }
-  if ('deployment' in newContractsConfig) {
-    if ('accounts' in newContractsConfig.deployment) {
-      for (var i = 0; i < newContractsConfig.deployment.accounts.length; i++) {
-        newContractsConfig.deployment.accounts[i].balance = utils.getWeiBalanceFromString(newContractsConfig.deployment.accounts[i].balance, web3);
-      }
-    }
+  if ('accounts' in newContractsConfig.deployment) {
+    newContractsConfig.deployment.accounts.forEach((account, index) => {
+      account.balance = utils.getWeiBalanceFromString(account.balance, web3);
+    });
```

Comment:
Oh look, how wonderful!

Okay we still need some feedback from someone on the @embark-framework/core-devs who can confirm that there's always a `deployment` property in the contracts config.

Reply:
There is none of our templates or example that doesn't have it, though, to be sure, we  could just check for it just in case.